google/cloud/storage/client.cc

```diff
@@ -92,31 +92,30 @@ integrity checks using the DisableMD5Hash() and DisableCrc32cChecksum() options.
   // class checks before calling it.
   std::uint64_t source_size = google::cloud::internal::file_size(file_name);
 
-  auto result = UploadStreamResumable(source, source_size, request);
-  if (not result.first.ok()) {
-    google::cloud::internal::RaiseRuntimeError(result.first.error_message());
-  }
-  return result.second;
+  return UploadStreamResumable(source, source_size, request).value();
 }
 
-std::pair<Status, ObjectMetadata> Client::UploadStreamResumable(
+StatusOr<ObjectMetadata> Client::UploadStreamResumable(
     std::istream& source, std::uint64_t source_size,
     internal::ResumableUploadRequest const& request) {
-  Status status;
-  std::unique_ptr<internal::ResumableUploadSession> session;
-  std::tie(status, session) = raw_client()->CreateResumableSession(request);
-  if (not status.ok()) {
-    return std::make_pair(status, ObjectMetadata{});
+  StatusOr<std::unique_ptr<internal::ResumableUploadSession>> session_status =
+      raw_client()->CreateResumableSession(request);
+  if (not session_status.ok()) {
+    return std::move(session_status).status();
   }
 
+  auto session = *std::move(session_status);
```

Comment:
Might this be nicer as `auto session = std::move(*session_status)`?

Reply:
Fixed.